cpp/models/abm/person.h

```diff
@@ -385,24 +386,25 @@ class Person
     ScalarType get_mask_protective_factor(const Parameters& params) const;
 
     /**
-     * @brief For every #LocationType a Person has a compliance value between -1 and 1.
-     * -1 means that the Person never complies to any Mask duty at the given #LocationType.
-     * 1 means that the Person always wears a Mask a the #LocationType even if it is not required.
-     * @param[in] preferences The vector of Mask compliance values for all #LocationType%s.
+     * @brief For every #InterventionType a Person has a compliance value between -1 and 1.
+     * -1 means that the Person never complies to the Intervention.
+     * 1 means that the Person always complies to the Intervention even if it is not required.
+     * @param[in] intervention_type The intervention type. 
+     * @param[in] value The compliance value.
      */
-    void set_mask_preferences(std::vector<ScalarType> preferences)
+    void set_compliance(InterventionType intervention_type, ScalarType value)
     {
-        m_mask_compliance = preferences;
+        m_compliance[static_cast<int>(intervention_type)] = value;
```

Comment:
Please use the correct underlying type for these casts, i.e. uint32_t, even though it probably won't cause any issues to use int.

Reply:
It's fixed!